client/components/parts/ButtonField.jsx

```diff
@@ -33,13 +34,56 @@ export default class ButtonField extends Component{
       'facebook',
       'twitter',
       'instagram'
-    ]
+    ],
+    typeTemplates: {
+      link: {
+        label: 'Link URL',
+        placeholder: 'http://www.somewhere.com'
+      },
+      email: {
+        label: 'Email Address',
+        preprefix: 'mailto:',
+        placeholder: 'person@somewhere.com'
+      },
+      facebook: {
+        label: 'Facebook URL',
+        placeholder: 'group/something',
+        preprefix: 'http://',
+        prefix: 'facebook.com/'
+      },
+      twitter: {
+        label: 'Twitter Handle',
+        placeholder: 'someone',
+        preprefix: 'http://',
+        prefix: 'twitter.com/'
+      },
+      instagram: {
+        label: 'Instagram Account',
+        placeholder: 'something',
+        preprefix: 'http://',
+        prefix: 'instagram.com/'
+      }
+    }
   }
 
   handleChange = (e) => {
     const kind = e.target.dataset.kind
     const val = this.props.value
-    val[kind] = e.target.value
+    if (kind === 'type'){
+      // Swap prefixes
+      const oldtemplate = this.props.typeTemplates[val.type]
+      const newtemplate = this.props.typeTemplates[e.target.value]
+      val.href = val.href.replace(new RegExp(`^(${oldtemplate.preprefix}|${oldtemplate.prefix})+`), '')
+      if (newtemplate.prefix) val.href = newtemplate.prefix + val.href
+      if (newtemplate.preprefix) val.href = newtemplate.preprefix + val.href
```

Comment:
Add some comments here explaining what's going on -- hard to understand the purpose of pre/prepre-fixes and the regex